### PR TITLE
Apply k8s 1.18.5

### PR DESCRIPTION
**Pull request by tomahawk28**
Towards https://github.com/giantswarm/giantswarm/issues/12180

## Checklist

- [x] Update changelog in CHANGELOG.md.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -16,9 +16,9 @@ require (
 	github.com/spf13/afero v1.3.2
 	github.com/ulikunitz/xz v0.5.7 // indirect
 	gopkg.in/yaml.v2 v2.3.0
-	helm.sh/helm/v3 v3.1.3
-	k8s.io/apimachinery v0.17.2
-	k8s.io/client-go v0.17.2
+	helm.sh/helm/v3 v3.2.4
```
Comment (rossf7) [inline go.mod:19]: 🎉  Looks good. Thx!

File: go.mod [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
+module github.com/giantswarm/helmclient/v2
 
 go 1.14
 
 require (
 	github.com/dsnet/compress v0.0.1 // indirect
 	github.com/giantswarm/backoff v0.2.0
-	github.com/giantswarm/k8sclient/v3 v3.1.2
-	github.com/giantswarm/kubeconfig v0.2.1
+	github.com/giantswarm/k8sclient/v4 v4.0.0-20200806115259-2d3b230ace59
```
Comment (rossf7) [inline go.mod:8]: TODO: Remove before merging.

File: go.mod [hunk 3]
```diff
@@ -1,12 +1,12 @@
...
+	github.com/giantswarm/kubeconfig/v2 v2.0.0-20200806114529-9ef53912cf03
```
Comment (rossf7) [inline go.mod:9]: TODO: Remove before merging.

File: go.mod [hunk 4]
```diff
@@ -16,9 +16,9 @@ require (
...
 	gopkg.in/yaml.v2 v2.3.0
-	helm.sh/helm/v3 v3.1.3
-	k8s.io/apimachinery v0.17.2
-	k8s.io/client-go v0.17.2
+	helm.sh/helm/v3 v3.2.4
+	k8s.io/apimachinery v0.18.5
+	k8s.io/client-go v0.18.5
 	rsc.io/letsencrypt v0.0.3 // indirect
-	sigs.k8s.io/controller-runtime v0.4.0
+	sigs.k8s.io/controller-runtime v0.6.1
```
Comment (xh3b4sd) [inline go.mod:23]: Are we sure we can go that far? I remember there were breaking changes and that may just causes more disruption in the operators. If the new k8s deps sort this then this is super cool. Just wanted to raise a thought here.

Comment (rossf7) [reply go.mod:23]: This was outdated. In operatorkit we're using v0.5.2. But we need to use v0.6.X for K8s 1.18 support.

We used v0.6.1 in all the test projects and it has been OK.

**Pull request merged**